chore: improve guide (#193)
babylonchain/networks: bbn-test-4/finality-providers/README.md

and will not be counted towards the active stake of the system.
Note that the deposit is still fully in the custody of the creator of the
transaction, but will only become unlocked after the deposit period expires.

> The deposit amount of `10000000` signet satoshi is the minimum amount required

Reviewer: Let's highlight this with a warning etc?

Author: like this? https://github.com/babylonchain/networks/pull/193/commits/85c62fa0c8cbfa062b7b8cce14636dd841f39571

Note that the deposit is still fully in the custody of the creator of the
transaction, but will only become unlocked after the deposit period expires.

**⚠ Warning!**
The deposit amount of `10000000` signet satoshi is the minimum amount required
for registration, any deposit bellow this number will be considered an **invalid
registration**. There is no need to deposit more than `10000000` either, but if
it happens to exist a deposit tx with a higher value, it will be considered

> a valid registration.

Author: https://github.com/babylonchain/networks/pull/193/commits/a382b6949d5b7343ecc998a38f233aa8b156dd15

The deposit is a Bitcoin transaction with an output containing the deposit value
committing to the Babylon Bitcoin Staking script.
A special set of values should be used for the deposit to be a valid one.
More specifically,
to create a valid deposit, you can follow the steps
[in this guide](https://github.com/babylonchain/btc-staker/blob/ebb2e845d7be835c7d7f7e0347600092e5de67a3/docs/create-phase1-staking.md),
with the following flags on the
`stakercli transaction create-phase1-staking-transaction` command:

- `--finality-provider-pk=<fp_pk>` The public key of your finality provider
previous generated.

> - `--staker-pk` The public key of the account who has funds.

Author: https://github.com/babylonchain/networks/pull/193/commits/a382b6949d5b7343ecc998a38f233aa8b156dd15

```shell
stakercli transaction create-phase1-staking-transaction \

>   --staker-pk <your_generated_staker_pub_key> \

Reviewer: Let's also add a note that (1) it can be different for the finality provider pub key or the same, depending on where the funds have been sent and (2) it is recommended to keep the finality provider keyring offline, so use a separate account to create a transaction?

Author: https://github.com/babylonchain/networks/pull/193/commits/416ffd8668f420f8c19aeba8d44144e06ccd2b49

- `--covenant-quorum=1`
- `--network=signet`

> The public keys used in the flags `--staker-pk` and `--finality-provider-pk`
> can be different from each other or the same, depends on the setup and where
> the funds have been sent. Although, is recommended to keep the finality
> provider keyring offline, so try to use a separate account to create the
> transaction.

Reviewer: ```suggestion
The difference between `--staker-pk` and `--finality-provider-pk`
is that the `--finality-provider-pk` flag specifies the public key of the
finality provider being registered, while the `--staker-pk` flag specifies
the Bitcoin public key that funds the deposit transaction.
It is recommended that the finality provider public key should be securely
stored and not hold any funds. Therefore, for creating the deposit
transaction, it is recommended that one funds the deposit transaction
with a separate public key (i.e. the `--staker-pk`) that holds the funds.
```

}
```

After signing the transaction you should have it in
[hex format](https://github.com/babylonchain/btc-staker/blob/ebb2e845d7be835c7d7f7e0347600092e5de67a3/docs/create-phase1-staking.md#sign-transaction).
It is possible to verify if your transaction has the correct parameters before

> submit the transaction to BTC chain by using the following script:

Reviewer: ```suggestion
submitting it to the Signet BTC ledger by using the following script:
```

Author: https://github.com/babylonchain/networks/pull/193/commits/a382b6949d5b7343ecc998a38f233aa8b156dd15

```shell

> export FP_BTC_PK=<your_fp_pk>
> export SIGNED_TX=<your-signed-tx-hex>
> 
> ./bbn-test-4/finality-providers/scripts/fp-check-tx.sh

Reviewer: ```suggestion
$ export FP_BTC_PK=<your_fp_pk>
$ export SIGNED_TX=<your-signed-tx-hex>

$ ./bbn-test-4/finality-providers/scripts/fp-check-tx.sh
```

Author: https://github.com/babylonchain/networks/pull/193/commits/a382b6949d5b7343ecc998a38f233aa8b156dd15

Provided transaction is valid staking transaction!
```

> This signed tx in hex format is ready for propagation to the Bitcoin ledger,
> which can happen in several ways:

Reviewer: ```suggestion
The success of the above command means that the signed transaction in hex format
is ready for propagation to the Bitcoin ledger.
This can happen in several ways:
```

Author: https://github.com/babylonchain/networks/pull/193/commits/a382b6949d5b7343ecc998a38f233aa8b156dd15

but with the `.sig` extension (e.g. `${nickname}.sig`).
The content of the file should be the plain value of the `schnorr_signature_hex` field.

The signature was generated by reading the entire file data, not only the file

> content. For proper verification, the exactly file used for signing should

Reviewer: ```suggestion
content. For proper verification, the exact file used for signing should
```

Let's also highlight this with a warning etc?

Author: https://github.com/babylonchain/networks/pull/193/commits/8b52ad3dcd8691a2934708490b1f24e95460d8bd

Author: https://github.com/babylonchain/networks/pull/193/commits/e680ef1cba6f593c4a4907264b45a07c0902e95a

be submited in the pull request.

## 5. Create Pull Request

Submit your finality provider information under the `registry` directory and
your signature under the `sigs` directory. Both file names should have the
same name (e.g. `${nickname}`), but with `.json` and `.sig` extensions respectively.
**Make sure that you submit exactly the same file that you signed to ensure proper
verification**.

> If you have installed all the binaries in your path, check out locally if your
> finality provider is valid before creating the pull request:

Reviewer: ```suggestion
If you have installed all the binaries in your path, you can check out locally if your
finality provider is valid before creating the pull request,
by using the following script contained in this repo
```

Reviewer: We should also have instructions for verifying that the deposit transaction is valid before sending it to BTC.

Author: I can split the verification into multiple files

Author: https://github.com/babylonchain/networks/pull/193/commits/0195703c03ffc6325ed9f286db69b230871197e7

but with the `.sig` extension (e.g. `${nickname}.sig`).
The content of the file should be the plain value of the `schnorr_signature_hex` field.

**⚠ Warning!**
The signature was generated by reading the entire file data, not only the file
content. For proper verification, the exact file used for signing should
be submited in the pull request.

## 5. Create Pull Request

Submit your finality provider information under the `registry` directory and
your signature under the `sigs` directory. Both file names should have the
same name (e.g. `${nickname}`), but with `.json` and `.sig` extensions respectively.

> **Make sure that you submit exact the same file that you signed to ensure proper

Reviewer: ```suggestion
**Make sure that you submit exactly the same file that you signed to ensure proper
```

Author: https://github.com/babylonchain/networks/pull/193/commits/a382b6949d5b7343ecc998a38f233aa8b156dd15
